smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/endpoints/EndpointResolverGenerator.kt

```diff
@@ -9,111 +9,59 @@ import software.amazon.smithy.codegen.core.Symbol
 ) {
     fun render(ctx: ProtocolGenerator.GenerationContext) {
         val rootNamespace = ctx.settings.moduleName
 
         val ruleSetNode = ctx.service.getTrait<EndpointRuleSetTrait>()?.ruleSet
         val ruleSet = if (ruleSetNode != null) EndpointRuleSet.fromNode(ruleSetNode) else null
 
-        ctx.delegator.useFileWriter("./$rootNamespace/EndpointResolver.swift") {
-            it.addImport(dependency.target)
-            val endpointParamsGenerator = EndpointParamsGenerator(ruleSet)
-            endpointParamsGenerator.render(it, ctx)
-        }
-
-        ctx.delegator.useFileWriter("./$rootNamespace/EndpointResolver.swift") {
+        ctx.delegator.useFileWriter("./$rootNamespace/Endpoints/EndpointResolver.swift") {
             renderResolverProtocol(it)
             it.write("")
             renderResolver(it, ruleSet)
             val inputSymbol = Symbol.builder().name("SdkHttpRequestBuilder").build()
             val outputSymbol = Symbol.builder().name("OperationStackOutput").build()
             val outputErrorSymbol = Symbol.builder().name("OperationStackError").build()
-            it.addImport(SwiftDependency.CLIENT_RUNTIME.target)
-            it.addImport(SwiftDependency.SMITHY_HTTP_API.target)
             it.write("")
-            MiddlewareGenerator(it, endpointResolverMiddleware(it, inputSymbol, outputSymbol, outputErrorSymbol))
-                .generate()
         }
     }
 
     private fun renderResolverProtocol(writer: SwiftWriter) {
-        writer.openBlock("public protocol \$L {", "}", EndpointTypes.EndpointResolver) {
-            writer.write("func resolve(params: EndpointParams) throws -> \$L", SmithyHTTPAPITypes.Endpoint)
+        writer.openBlock("public protocol \$N {", "}", EndpointTypes.EndpointResolver) {
+            writer.write("func resolve(params: EndpointParams) throws -> \$N", SmithyHTTPAPITypes.Endpoint)
         }
     }
 
-    private fun renderResolver(writer: SwiftWriter, endpointRules: EndpointRuleSet?) {
-        writer.openBlock("public struct \$L: \$L  {", "}", EndpointTypes.DefaultEndpointResolver, EndpointTypes.EndpointResolver) {
-            writer.write("")
-            endpointRules?.let {
-                writer.write("private let engine: \$L", ClientRuntimeTypes.Core.EndpointsRuleEngine)
-                writer.write("private let ruleSet = \$S", Node.printJson(endpointRules.toNode()))
-            }
-            writer.write("")
-            writer.openBlock("public init() throws {", "}") {
-                endpointRules?.let {
-                    writer.write("engine = try \$L(partitions: \$L, ruleSet: ruleSet)", ClientRuntimeTypes.Core.EndpointsRuleEngine, partitionDefinition)
-                }
+    private fun renderResolver(writer: SwiftWriter, endpointRuleSet: EndpointRuleSet?) {
+        writer.write(
+            "typealias DefaultEndpointResolver = \$N<EndpointParams>",
+            ClientRuntimeTypes.Core.DefaultEndpointResolver,
+        )
```

Comment:
The definition of the endpoint resolver for a service is just a typealias of the reusable endpoint resolver with the service's params type.